src/consensus/aft/test/driver.h

```diff
@@ -302,4 +486,88 @@ class RaftDriver
       }
     }
   }
+
+  void assert_state_sync()
+  {
+    auto [target_id, nd] = *_nodes.begin();
+    auto& target_raft = nd.raft;
+    const auto target_term = target_raft->get_term();
+    const auto target_last_idx = target_raft->get_last_idx();
+    const auto target_commit_idx = target_raft->get_commit_idx();
+
+    const auto target_final_entry =
+      target_raft->ledger->get_entry_by_idx(target_last_idx);
+
+    bool all_match = true;
+    for (auto it = std::next(_nodes.begin()); it != _nodes.end(); ++it)
+    {
+      const auto& node_id = it->first;
+      auto& raft = it->second.raft;
+
+      if (raft->get_term() != target_term)
+      {
+        RAFT_DRIVER_OUT
+          << fmt::format(
+               "  Note over Node{}: Term {} doesn't match term {} on Node{}",
+               node_id,
+               raft->get_term(),
+               target_term,
+               target_id)
+          << std::endl;
+        all_match = false;
+      }
```

Comment:
These were asserts (hence the name `assert_state_sync` for this step), but I realised we really want them to trigger a failure in non-Debug builds, and to produce a verbose description of the failure. So now if any of these conditions aren't met, we output a note in the Mermaid diagram describing the mismatch, which is really helpful in analysing bad traces.